### PR TITLE
Extend segmentation

### PR DESCRIPTION
**Pull request by jwboth**
Simple modifications to segmentation:
- median of scikit-image internally uses conversion to np.uint8 if not provided and drops a warning. The conversion is now performed in daria to surpress the warning.
- Keyword is added to allow for different monochromatic bases for the segmentation (other than just grayscale images).

### CODE REVIEW

File: src/daria/utils/segmentation.py [hunk 1]
```diff
@@ -40,30 +40,46 @@ def segment(
...
         raise ValueError(f"img of type {type(img)} not supported.")
 
+    # Require scalar representation - the most natural general choice is either to
+    # use a grayscale representation or the value component of the HSV version,
+    # when.
+    monochromatic = kwargs.pop("monochromatic_color", "gray")
+    if monochromatic == "gray":
+        basis = cv2.cvtColor(basis, cv2.COLOR_RGB2GRAY)
+    elif monochromatic == "value":
+        hsv = cv2.cvtColor(img, cv2.COLOR_RGB2HSV)
```
Comment (EStorvik) [inline src/daria/utils/segmentation.py:67]:
```suggestion
        hsv = cv2.cvtColor(basis, cv2.COLOR_RGB2HSV)
```

Comment (EStorvik) [reply src/daria/utils/segmentation.py:67]: I guess img still can be a darsia image?

**Review by EStorvik — APPROVED** [review]
Looks good, just check the comment!

File: src/daria/utils/segmentation.py (hunk 1 above)
Comment (jwboth) [reply src/daria/utils/segmentation.py]: Yes.

**Pull request merged**